webpush/push.go

```diff
@@ -69,6 +69,42 @@ func NewPushRequest(sub *Subscription, message string, token string) (*http.Requ
 	return req, nil
 }
 
+// NewVapidRequest creates a valid Web Push HTTP request for sending a message
+// to a subscriber, using Vapid authentication. You can add more headers to
+// configure collapsing, TTL.
+func NewVapidRequest(sub *Subscription, message string, vapid *Vapid) (*http.Request, error) {
```

Comment:
If possible I'd like NewVapidRequest to be a thin wrapper around NewPushRequest. Is it feasible to call `NewPushRequest` and then manipulate the headers rather than duplicating the code?